go/vt/vttablet/endtoend/views_test.go

```diff
@@ -153,6 +152,77 @@ func TestDropViewDDL(t *testing.T) {
 	require.Zero(t, qr.Rows)
 }
 
+// TestViewDDLWithInfrSchema will validate information schema queries with views.
+func TestViewDDLWithInfrSchema(t *testing.T) {
```

Comment:
We should perhaps add an E2E tests where we try to add a view with two different keyspaces in the select query and look for `VT12001` error

Reply:
we have a unit test in plan_test that covers it.
views_test is vttablet e2e test, the failure will happen in vtgate while planning.